lighthouse-cli/bin.ts

```diff
@@ -35,10 +35,12 @@ import * as Printer from './printer';
 import * as randomPort from './random-port';
 import {Results} from './types/types';
 const yargs = require('yargs');
-const opn = require('opn');
-const updateNotifier = require('update-notifier');
 const pkg = require('../package.json');
 
+// accept noop modules for these, so the real dependency is optional.
+const opn = require('./shim-modules').opn();
```

Comment:
the try/catch in the same file is really nasty and i really preferred moving it out into shim-modules.

so i think it's worth keeping that.

However, i'd be fine with guarding the calls so we don't need to mock their API.